src/vs/editor/contrib/links/links.ts

```diff
@@ -295,31 +296,48 @@ export class LinkDetector implements IEditorContribution {
 		const { link } = occurrence;
 
 		link.resolve(CancellationToken.None).then(uri => {
+			let uriToUse = uri;
 
 			// Support for relative file URIs of the shape file://./relativeFile.txt or file:///./relativeFile.txt
-			if (typeof uri === 'string' && this.editor.hasModel()) {
+			// Support for workspace://<workspace-name>/path/to/file and project://path/to/file (and relative project path)
+			if (typeof uriToUse === 'string' && this.editor.hasModel()) {
 				const modelUri = this.editor.getModel().uri;
-				if (modelUri.scheme === Schemas.file && strings.startsWith(uri, 'file:')) {
-					const parsedUri = URI.parse(uri);
-					if (parsedUri.scheme === Schemas.file) {
-						const fsPath = resources.originalFSPath(parsedUri);
-
-						let relativePath: string | null = null;
-						if (strings.startsWith(fsPath, '/./')) {
-							relativePath = `.${fsPath.substr(1)}`;
-						} else if (strings.startsWith(fsPath, '//./')) {
-							relativePath = `.${fsPath.substr(2)}`;
-						}
 
-						if (relativePath) {
-							uri = resources.joinPath(modelUri, relativePath);
+				// strip name part of link now that {@link https://google.com|Google is allowed}
```

Comment:
This is VS Code's core link detection logic which is pretty language agnostic. I'm not sure this change belongs here (especially because some valid urls do include `|`)

Instead, we would likely want to handle this in the TypeScript extension

Reply:
The "|" character is considered "unsafe" according to the RFC and would have to be encoded.   I do believe this is the reasoning around this character being used to separate URL from the label with jsdoc.

https://www.ietf.org/rfc/rfc1738.txt

```
   Characters can be unsafe for a number of reasons.  The space
   character is unsafe because significant spaces may disappear and
   insignificant spaces may be introduced when URLs are transcribed or
   typeset or subjected to the treatment of word-processing programs.
   The characters "<" and ">" are unsafe because they are used as the
   delimiters around URLs in free text; the quote mark (""") is used to
   delimit URLs in some systems.  The character "#" is unsafe and should
   always be encoded because it is used in World Wide Web and in other
   systems to delimit a URL from a fragment/anchor identifier that might
   follow it.  The character "%" is unsafe because it is used for
   encodings of other characters.  Other characters are unsafe because
   gateways and other transport agents are known to sometimes modify
   such characters. These characters are "{", "}", "|", "\", "^", "~",
   "[", "]", and "`".
```

Reply:
Please split this into its own issue then. It is not js/ts specific should likely be fixed here: https://github.com/mjbvz/vscode/blob/5bcfb8e897669f0069c73c69506d6371dd7ae3d8/src/vs/editor/common/modes/linkComputer.ts#L276

(Although again, I seem to recall some links using `|` in their query strings)